docs/extensions/flatten-history.py

```diff
@@ -0,0 +1,175 @@
+import os
+import re
+import pathlib
+
+from docutils.parsers.rst import Directive
+from docutils.parsers.rst import directives
+from docutils.statemachine import ViewList
+from sphinx.util.nodes import nested_parse_with_titles
+from docutils import nodes
+
+# Nice variable references for parts of
+# a version tuple (MAJOR, MINOR, RELEASE) ex: (24.3.1)
+MAJOR = 0
+MINOR = 1
+RELEASE = 2
+
+# Supported history entry types
+HISTORY_TYPES = ["bug", "bugs", "feature", "features"]
+
+class FlattenVersionHistory(Directive):
+    """Custom directive to read mega version history files and flatten into one list"""
+
+    # Directive setup
+    require_arguments = 1 # Requires an entry type (see HISTORY_TYPES)
+    optional_arguments = 10 # Up to 10 versions to parse from
```

Comment:
So for 24.9.3., the version-history.rst file will look something like this, right?
```
.. include-history:: features 24.9.3 23.19.3
# ....
.. include-history:: bugs 24.9.3. 24.0.9 23.19.3 23.0.19
```
which assumes that we release new current-feature versions no less frequently than we release bug fixes for any of the three previous series.  I guess since we won't be including 24.x more than once in bug list, 10 entries will be enough for a current-feature release that took so long to get out the door that we released three full sets of bugfixes in the interim.

So I've convinced myself that 10 is probably enough, but -- will Sphinx warn if the list goes up to 11?  Silently ignoring a version number would not be OK.

Reply:
I did some local testing and can confirm the following:
- Only 9 versions can be specified because the required type is included in the count
- If the number of arguments is exceeded then sphinx prints an error and nothing is displayed in the docs

Reply:
Also, the above example appears correct. I did convert a single recent small version history as an example. Obviously it doesn't seem as nice since it is only flattening one section into itself.